From 86ed1112990f92bc184c80183d1a7c709fc80868 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 20 Oct 2015 17:38:50 +0100
Subject: [PATCH] Do not enumerate the default device when we have PulseAudio
 running, or the duplicate sink will cause some confusion. Use a friendlier
 name when PulseAudio is running.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index 4d87afa..0886020 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -1572,6 +1572,8 @@ void CAESinkALSA::EnumerateDevice(AEDeviceInfoList &list, const std::string &dev
 
   snd_pcm_close(pcmhandle);
   info.m_wantsIECPassthrough = true;
+  if (info.m_displayName == "PulseAudio Sound Server")
+        info.m_displayName.replace(info.m_displayName.begin(), info.m_displayName.end(), "OSMC streaming to Bluetooth speaker / headphones");
   list.push_back(info);
 }
 
-- 
2.7.4

